iceoryx_hoofs/include/iceoryx_hoofs/cxx/expected.hpp

```diff
@@ -453,6 +453,39 @@ class IOX_NO_DISCARD expected<void, ErrorType> : public expected<ErrorType>
     using expected<ErrorType>::expected;
 };
 
+/// @brief equality check for two distinct expected types
+/// @tparam ErrorType type of the error stored in the expected
+/// @param[in] lhs left side of the comparision
+/// @param[in] rhs right side of the comparision
+/// @return true if the expecteds are equal, otherwise false
+template <typename ErrorType>
+constexpr bool operator==(const expected<ErrorType>& lhs, const expected<ErrorType>& rhs);
```

Comment:
I think you forgot `noexcept` here and in the other operator implementations.

Reply:
I think equality operators can't be `noexcept` in this case as we call the equality operator of an underlying type inside the `expected`. Also in case of the `variant` the `if (N != index)` it could lead to an `Expects` call which could throw in future if configured that way by the user.